### PR TITLE
[core][tests] Add nightly test for datasets random_shuffle and sort

### PR DESCRIPTION
**Pull request by stephanie-wang**
## Why are these changes needed?

Adding a large-scale nightly test for Datasets random_shuffle and sort. The test script generates random blocks and reports total run time and peak driver memory.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/dataset/sort.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    print(ds.stats())
+
+    if "TEST_OUTPUT_JSON" in os.environ:
+        out_file = open(os.environ["TEST_OUTPUT_JSON"], "w")
+        results = {
+                "time": end_time - start_time,
+                "success": "1" if exc is None else "0",
+                "num_partitions": num_partitions,
+                "partition_size": partition_size,
+                "perf_metrics": [
```
Comment (stephanie-wang) [inline release/nightly_tests/dataset/sort.py:151]: @jjyao I'll try running it, but can you also look at this and make sure I'm reporting the metrics correctly for the dashboard?

Comment (jjyao) [reply release/nightly_tests/dataset/sort.py:151]: Should we also have a perf metric for runtime?

**Review by jjyao — APPROVED** [review]

**Review by mwtian — APPROVED** [review]

File: release/nightly_tests/dataset/sort.py (hunk 1 above)
Comment (stephanie-wang) [reply release/nightly_tests/dataset/sort.py:151]: Ah yes, I didn't realize the total run time wouldn't be enough. What do you put for the metric type in that case?

Comment (jjyao) [reply release/nightly_tests/dataset/sort.py:151]: "LATENCY"

**Review by jjyao — APPROVED** [review]

**Comment by stephanie-wang** [comment]
Hmm weirdly it failed with OOM on the random_shuffle version [here](https://buildkite.com/ray-project/release-tests-branch/builds/359#211d782e-1f83-4fb3-9371-a48b47ea5a10) but not on the sort version.

But now I'm running it manually and it seems to be passing successfully. Maybe I'll downgrade to 1k partitions for now and see if that passes.

**Pull request merged**